lib/oppen/scan_stack.rb

```diff
@@ -112,5 +114,15 @@ def pop_bottom
       end
       res
     end
+
+    # Offset the values of the stack.
+    # @param offset [Integer]
+    #
+    # @return [Nil]
+    def update_indexes(offset)
+      @stack = @stack.map { |val|
+        (val + offset) % length if val
+      }
+    end
```

Comment:
Are you sure it returns `nil`. I am almost certain the return type is that of `@stack`. Verify please.